### DISCUSSION TITLE
Gh pages hasn't deployed my create-react-app and react-router app

### DISCUSSION THREAD

**Discussion by KarolChilimoniuk** [community/community#23240] [Other Feature Feedback, Questions, & Ideas]
<p>Hello everybody, I am new in this community and finally it is this time when I have to ask you for your help. I would like to deploy my practice app in gh-pages. I read instructions how to do it and… it didn’t work. I still looking for a mistake I did but I can’t find it :confused:</p>
<p>In dev tools console i received:<br>
Failed to load resource: the server responded with a status of 404 ()<br>
2.01830e90.chunk.js:1 Failed to load resource: the server responded with a status of 404 ()<br>
main.9bb4cdd3.chunk.js:1 Failed to load resource: the server responded with a status of 404 ()<br>
2.01830e90.chunk.js:1 Failed to load resource: the server responded with a status of 404 ()<br>
main.9bb4cdd3.chunk.js:1 Failed to load resource: the server responded with a status of 404 ()<br>
(index):1 Unchecked runtime.lastError: The message port closed before a response was received.<br>
manifest.json:1 Failed to load resource: the server responded with a status of 404 ()<br>
manifest.json:1 Manifest: Line: 1, column: 1, Syntax error.<br>
main.5646360f.chunk.css:1 Failed to load resource: the server responded with a status of 404 ()<br>
manifest.json:1 Failed to load resource: the server responded with a status of 404 ()</p>
<p>This is link to my github repo of my project:</p>
<p><a href="https://github.com/KarolChilimoniuk/ReminderApp" rel="noopener nofollow ugc">project repo</a></p>
<p>Thank you in advance for your help</p>

**Comment by otter-computer** [comment, marked as answer]
<p>Hi <a class="mention" href="/karolchilimoniuk">@KarolChilimoniuk</a>, welcome to the GitHub Support Community! I think this is the same issue as another user was having in this thread here:</p>
<aside class="quote quote-modified" data-post="3" data-topic="142838">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/gh-pages-404-errors-with-deploy-a-react-app/142838/3">Gh-pages 404 errors with deploy a react app</a> <a class="badge-wrapper  bullet" href="/c/github-help/github-pages/31"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #c8e1ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Pages—websites for you and your projects hosted right on your GitHub repository.">GitHub Pages</span></a>
  </div>
  <blockquote>
    Hi <a class="mention" href="/xahtc">@XAHTC</a>, welcome to the GitHub Support Community! 
I’m not seeing those errors in your Pages site anymore, however it looks like they were caused by your asset URLs missing the test-app repository name section. 
When you’re deploying a React app to GitHub Pages there’s some steps you need to follow to make sure that everything works correctly—specifically adding the homepage variable to your package.json file. This means that all your URLs include the required repo name. 
For exact steps I’d r…
  </blockquote>
</aside>

<p>It looks like your asset links are missing the <code>ReminderApp</code> section of the URL. You need to follow a few extra steps to get your React app working on Pages, including filling out the <code>homepage</code> variable in your <code>package.json</code>. I’d recommend reading over the instructions here:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://create-react-app.dev/docs/deployment/#github-pages">
  <header class="source">
      <img src="https://create-react-app.dev/img/favicon/favicon.ico" class="site-icon" width="64" height="64">

      <a href="https://create-react-app.dev/docs/deployment/#github-pages" target="_blank" rel="noopener">create-react-app.dev</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://user-images.githubusercontent.com/2439407/181075716-6a50c18b-b23d-42e0-aec8-8202bd665dd0.png" class="thumbnail" width="690" height="362" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/8/2/8223ad9697262791355b25ca7df8b372d92969eb_2_10x10.png"></div>

<h3><a href="https://create-react-app.dev/docs/deployment/#github-pages" target="_blank" rel="noopener">Deployment | Create React App</a></h3>

  <p>npm run build creates a build directory with a production build of your app. Set up your favorite HTTP server so that a visitor to your site is served index.html, and requests to static paths like /static/js/main..js are served with the contents of...</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by otter-computer** [comment]
<p>I just noticed you <em>have</em> added the <code>homepage</code> variable, but you’ve added it inside the <code>dependencies</code> object instead of on the top level. You’ll need to move that outside of <code>dependencies</code> and into the same level as <code>name</code>.</p>

**Comment by KarolChilimoniuk** [comment]
<p>Sorry for my late response but my Christmas vacation was longer than I supposed :] Thank you for your help. I realised that I made a mistake in my package.json and additionally I read create-react-app documentation carefully in Deployment chapter and I have found a solution of my problem.</p>
